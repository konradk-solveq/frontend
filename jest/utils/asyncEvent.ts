import {act} from '@testing-library/react-native';

const asyncEvent = async (event: any) => {
    await act(async () => {
        await event;
    });
};

export default asyncEvent;
