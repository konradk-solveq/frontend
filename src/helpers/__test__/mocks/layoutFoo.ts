export const centerLeftPxResult = {
    small: [372.2826086956522, 364.1304347826087, 347.82608695652175],
    medium: [293.47826086956525, 266.304347826087, 230.0724637681159],
    big: [121.37681159420289, 76.08695652173914, 12.681159420289854],
};

export const centerTopPxResult = {
    small: [665.5111607142857, 661.0446428571429, 655.0892857142857],
    medium: [585.1138392857143, 555.3370535714286, 525.5602678571429],
    big: [354.34375, 175.68303571428572, 71.46428571428571],
};

export const horizontalResult = {
    small: ['0.725%', '2.899%', '7.246%'],
    medium: ['21.739%', '28.986%', '38.647%'],
    big: ['67.633%', '79.710%', '96.618%'],
};

export const horizontalPxResult = {
    small: [5.434782608695652, 21.73913043478261, 54.34782608695652],
    medium: [163.04347826086956, 217.3913043478261, 289.8550724637681],
    big: [507.24637681159425, 597.8260869565217, 724.6376811594203],
};

export const verticalResult = {
    small: ['0.223%', '0.893%', '1.786%'],
    medium: ['12.277%', '16.741%', '21.205%'],
    big: ['46.875%', '73.661%', '89.286%'],
};

export const verticalPxResult = {
    small: [2.9776785714285716, 11.910714285714286, 23.821428571428573],
    medium: [163.77232142857144, 223.32589285714286, 282.8794642857143],
    big: [625.3125, 982.6339285714287, 1191.0714285714287],
};

export const stackHeaderHeightResult = 148.88392857142858;

export const widthResult = {
    small: ['0.725%', '2.899%', '7.246%'],
    medium: ['21.739%', '28.986%', '38.647%'],
    big: ['67.633%', '79.710%', '96.618%'],
};

export const widthOfResult = {
    small: ['0.725%', '2.899%', '7.246%'],
    medium: ['21.739%', '28.986%', '38.647%'],
    big: ['67.633%', '79.710%', '96.618%'],
};

export const widthPxResult = {
    small: [5.434782608695652, 21.73913043478261, 54.34782608695652],
    medium: [163.04347826086956, 217.3913043478261, 289.8550724637681],
    big: [507.2463768115942, 597.8260869565217, 724.6376811594203],
};

export const widthPxOfResult = {
    small: [5.434782608695652, 21.73913043478261, 54.34782608695652],
    medium: [163.04347826086956, 217.3913043478261, 289.8550724637681],
    big: [507.2463768115942, 597.8260869565217, 724.6376811594203],
};

export const heightResult = {
    small: ['0.223%', '0.893%', '1.786%'],
    medium: ['12.277%', '16.741%', '21.205%'],
    big: ['46.875%', '73.661%', '89.286%'],
};

export const heightPxResult = {
    small: [2.977678571428571, 11.910714285714285, 23.82142857142857],
    medium: [163.77232142857142, 223.32589285714283, 282.8794642857143],
    big: [625.3125, 982.6339285714287, 1191.0714285714287],
};

export const heightOfPxResult = {
    small: [2.977678571428571, 11.910714285714285, 23.82142857142857],
    medium: [163.77232142857142, 223.32589285714283, 282.8794642857143],
    big: [625.3125, 982.6339285714287, 1191.0714285714287],
};

export const relativeWidthResult = {
    small: ['0.596%', '2.382%', '5.955%'],
    medium: ['17.866%', '23.821%', '31.762%'],
    big: ['55.583%', '65.509%', '79.405%'],
};

export const relativeHeightResult = {
    small: ['0.272%', '1.086%', '2.173%'],
    medium: ['14.938%', '20.370%', '25.802%'],
    big: ['57.037%', '89.629%', '108.641%'],
};

export const standardResult = {
    small: [
        {
            height: '0.223%',
            left: 372.2826086956522,
            position: 'absolute',
            top: 1012.4107142857143,
            width: '0.725%',
        },
        {
            height: '0.893%',
            left: 364.1304347826087,
            position: 'absolute',
            top: 803.9732142857143,
            width: '2.899%',
        },
        {
            height: '1.786%',
            left: 347.82608695652175,
            position: 'absolute',
            top: 640.2008928571429,
            width: '7.246%',
        },
    ],
    medium: [
        {
            height: '12.277%',
            left: 293.47826086956525,
            position: 'absolute',
            top: 145.90625,
            width: '21.739%',
        },
        {
            height: '16.741%',
            left: 266.304347826087,
            position: 'absolute',
            top: 187.59375,
            width: '28.986%',
        },
        {
            height: '21.205%',
            left: 230.0724637681159,
            position: 'absolute',
            top: 253.10267857142858,
            width: '38.647%',
        },
    ],
    big: [
        {
            height: '46.875%',
            left: 121.37681159420289,
            position: 'absolute',
            top: 38.70982142857143,
            width: '67.633%',
        },
        {
            height: '73.661%',
            left: 76.08695652173914,
            position: 'absolute',
            top: 17.86607142857143,
            width: '79.710%',
        },
        {
            height: '89.286%',
            left: 12.681159420289854,
            position: 'absolute',
            top: 7.444196428571429,
            width: '96.618%',
        },
    ],
};

export const standardPxResult = {
    small: [
        {
            height: 2.977678571428571,
            left: 372.2826086956522,
            position: 'absolute',
            top: 1012.4107142857143,
            width: 5.434782608695652,
        },
        {
            height: 11.910714285714285,
            left: 364.1304347826087,
            position: 'absolute',
            top: 803.9732142857143,
            width: 21.73913043478261,
        },
        {
            height: 23.82142857142857,
            left: 347.82608695652175,
            position: 'absolute',
            top: 640.2008928571429,
            width: 54.34782608695652,
        },
    ],
    medium: [
        {
            height: 163.77232142857142,
            left: 293.47826086956525,
            position: 'absolute',
            top: 145.90625,
            width: 163.04347826086956,
        },
        {
            height: 223.32589285714283,
            left: 266.304347826087,
            position: 'absolute',
            top: 187.59375,
            width: 217.3913043478261,
        },
        {
            height: 282.8794642857143,
            left: 230.0724637681159,
            position: 'absolute',
            top: 253.10267857142858,
            width: 289.8550724637681,
        },
    ],
    big: [
        {
            height: 625.3125,
            left: 121.37681159420289,
            position: 'absolute',
            top: 38.70982142857143,
            width: 507.2463768115942,
        },
        {
            height: 982.6339285714287,
            left: 76.08695652173914,
            position: 'absolute',
            top: 17.86607142857143,
            width: 597.8260869565217,
        },
        {
            height: 1191.0714285714287,
            left: 12.681159420289854,
            position: 'absolute',
            top: 7.444196428571429,
            width: 724.6376811594203,
        },
    ],
};

export const perfectPxResult = {
    small: [
        {
            height: 2.977678571428571,
            left: 321.5892857142857,
            position: 'absolute',
            top: 1012.4107142857143,
            width: 5.434782608695652,
        },
        {
            height: 11.910714285714285,
            left: 294.7901785714286,
            position: 'absolute',
            top: 803.9732142857143,
            width: 21.73913043478261,
        },
        {
            height: 23.82142857142857,
            left: 259.0580357142857,
            position: 'absolute',
            top: 640.2008928571429,
            width: 54.34782608695652,
        },
    ],
    medium: [
        {
            height: 163.77232142857142,
            left: 83.375,
            position: 'absolute',
            top: 145.90625,
            width: 163.04347826086956,
        },
        {
            height: 223.32589285714283,
            left: 92.30803571428572,
            position: 'absolute',
            top: 187.59375,
            width: 217.3913043478261,
        },
        {
            height: 282.8794642857143,
            left: 104.21875,
            position: 'absolute',
            top: 253.10267857142858,
            width: 289.8550724637681,
        },
    ],
    big: [
        {
            height: 625.3125,
            left: 23.821428571428573,
            position: 'absolute',
            top: 38.70982142857143,
            width: 507.2463768115942,
        },
        {
            height: 982.6339285714287,
            left: 23.821428571428573,
            position: 'absolute',
            top: 17.86607142857143,
            width: 597.8260869565217,
        },
        {
            height: 1191.0714285714287,
            left: 23.821428571428573,
            position: 'absolute',
            top: 7.444196428571429,
            width: 724.6376811594203,
        },
    ],
};

export const posStaticHeightResult = {
    small: [
        {
            height: 3.6231884057971016,
            left: 372.2826086956522,
            position: 'absolute',
            top: '75.893%',
            width: 5.434782608695652,
        },
        {
            height: 14.492753623188406,
            left: 364.1304347826087,
            position: 'absolute',
            top: '60.268%',
            width: 21.73913043478261,
        },
        {
            height: 28.985507246376812,
            left: 347.82608695652175,
            position: 'absolute',
            top: '47.991%',
            width: 54.34782608695652,
        },
    ],
    medium: [
        {
            height: 199.2753623188406,
            left: 293.47826086956525,
            position: 'absolute',
            top: '10.938%',
            width: 163.04347826086956,
        },
        {
            height: 271.7391304347826,
            left: 266.304347826087,
            position: 'absolute',
            top: '14.063%',
            width: 217.3913043478261,
        },
        {
            height: 344.2028985507246,
            left: 230.0724637681159,
            position: 'absolute',
            top: '18.973%',
            width: 289.8550724637681,
        },
    ],
    big: [
        {
            height: 760.8695652173913,
            left: 121.37681159420289,
            position: 'absolute',
            top: '2.902%',
            width: 507.2463768115942,
        },
        {
            height: 1195.6521739130435,
            left: 76.08695652173914,
            position: 'absolute',
            top: '1.339%',
            width: 597.8260869565217,
        },
        {
            height: 1449.2753623188405,
            left: 12.681159420289854,
            position: 'absolute',
            top: '0.558%',
            width: 724.6376811594203,
        },
    ],
};

export const onlyPosResult = {
    small: [
        {left: 372.2826086956522, position: 'absolute', top: '75.893%'},
        {left: 364.1304347826087, position: 'absolute', top: '60.268%'},
        {left: 347.82608695652175, position: 'absolute', top: '47.991%'},
    ],
    medium: [
        {left: 293.47826086956525, position: 'absolute', top: '10.938%'},
        {left: 266.304347826087, position: 'absolute', top: '14.063%'},
        {left: 230.0724637681159, position: 'absolute', top: '18.973%'},
    ],
    big: [
        {left: 121.37681159420289, position: 'absolute', top: '2.902%'},
        {left: 76.08695652173914, position: 'absolute', top: '1.339%'},
        {left: 12.681159420289854, position: 'absolute', top: '0.558%'},
    ],
};

export const posAndWidResult = {
    small: [
        {
            left: 372.2826086956522,
            position: 'absolute',
            top: '75.893%',
            width: '0.725%',
        },
        {
            left: 364.1304347826087,
            position: 'absolute',
            top: '60.268%',
            width: '2.899%',
        },
        {
            left: 347.82608695652175,
            position: 'absolute',
            top: '47.991%',
            width: '7.246%',
        },
    ],
    medium: [
        {
            left: 293.47826086956525,
            position: 'absolute',
            top: '10.938%',
            width: '21.739%',
        },
        {
            left: 266.304347826087,
            position: 'absolute',
            top: '14.063%',
            width: '28.986%',
        },
        {
            left: 230.0724637681159,
            position: 'absolute',
            top: '18.973%',
            width: '38.647%',
        },
    ],
    big: [
        {
            left: 121.37681159420289,
            position: 'absolute',
            top: '2.902%',
            width: '67.633%',
        },
        {
            left: 76.08695652173914,
            position: 'absolute',
            top: '1.339%',
            width: '79.710%',
        },
        {
            left: 12.681159420289854,
            position: 'absolute',
            top: '0.558%',
            width: '96.618%',
        },
    ],
};

export const posWithMinHeightResult = {
    small: [
        {
            height: 10,
            left: 372.2826086956522,
            position: 'absolute',
            top: '75.893%',
            width: 5.434782608695652,
        },
        {
            height: 12,
            left: 364.1304347826087,
            position: 'absolute',
            top: '60.268%',
            width: 21.73913043478261,
        },
        {
            height: 23.82142857142857,
            left: 347.82608695652175,
            position: 'absolute',
            top: '47.991%',
            width: 54.34782608695652,
        },
    ],
    medium: [
        {
            height: 170,
            left: 293.47826086956525,
            position: 'absolute',
            top: '10.938%',
            width: 163.04347826086956,
        },
        {
            height: 223.32589285714283,
            left: 266.304347826087,
            position: 'absolute',
            top: '14.063%',
            width: 217.3913043478261,
        },
        {
            height: 282.8794642857143,
            left: 230.0724637681159,
            position: 'absolute',
            top: '18.973%',
            width: 289.8550724637681,
        },
    ],
    big: [
        {
            height: 625.3125,
            left: 121.37681159420289,
            position: 'absolute',
            top: '2.902%',
            width: 507.2463768115942,
        },
        {
            height: 982.6339285714287,
            left: 76.08695652173914,
            position: 'absolute',
            top: '1.339%',
            width: 597.8260869565217,
        },
        {
            height: 1191.0714285714287,
            left: 12.681159420289854,
            position: 'absolute',
            top: '0.558%',
            width: 724.6376811594203,
        },
    ],
};

export const fontSizeResult = {
    small: [2, 8, 16],
    medium: [110, 150, 190],
    big: [420, 660, 800],
};

export const some = {
    small: [0, 0, 0],
    medium: [0, 0, 0],
    big: [0, 0, 0],
};
