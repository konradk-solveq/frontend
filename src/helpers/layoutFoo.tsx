import {Dimensions} from 'react-native';

// funkcje wiliczające położenie elementów proporcje wg layoutu
// https://app.zeplin.io/project/5fbf658b936bbbb842e3c43c/dashboard?sid=60397836dc2eff2bfdde033a
const {width, height} = Dimensions.get('window');

const BASE_HEIGHT = 896;
const BASE_WIDTH = 414;
const BASE_STACK_HEADER_HEIGHT = 100;

let appSize: {
    width: number;
    height: number;
    ratio: number;
} | null;

let objSize: {
    width: number;
    height: number;
    ratio: number;
} | null;

const setAppSize = (w: number, h: number) => {
    appSize = {
        width: w,
        height: h,
        ratio: w / h,
    };
};

const initAppSize = () => {
    setAppSize(width, height);
};

const setObjSize = (w: number, h: number) => {
    objSize = {
        width: w,
        height: h,
        ratio: w / h,
    };
};

const showError = () => {
    if (!objSize) {
        console.error('objSize is not defined!');
    }
    if (!appSize) {
        console.error('appSize is not defined!');
    }
};

const getCenterLeftPx = () => {
    if (!objSize || !appSize) {
        showError();
        return 0;
    }
    let res: number =
        ((BASE_WIDTH - objSize.width) / 2 / BASE_WIDTH) * appSize.width;
    return res;
};

const getCenterTopPx = () => {
    if (!objSize || !appSize) {
        showError();
        return 0;
    }
    let res: number =
        ((BASE_HEIGHT - objSize.height) / 2 / BASE_HEIGHT) * appSize.height;
    return res;
};

const getHorizontal = (px: number) => {
    let res: number = (px / BASE_WIDTH) * 100;
    return res.toFixed(3) + '%';
};

const getHorizontalPx = (px: number) => {
    return (width / BASE_WIDTH) * px;
};

const getVertical = (px: number) => {
    let res: number = (px / BASE_HEIGHT) * 100;
    return res.toFixed(3) + '%';
};

const getVerticalPx = (px: number) => {
    return (height / BASE_HEIGHT) * px;
};

const getStackHeaderHeight = () => {
    return getVerticalPx(BASE_STACK_HEADER_HEIGHT);
};

const getWidth = () => {
    if (!objSize) {
        showError();
        return '0%';
    }
    let res: number = (objSize.width / BASE_WIDTH) * 100;
    return res.toFixed(3) + '%';
};

const getWidthPx = () => {
    if (!objSize || !appSize) {
        showError();
        return 0;
    }
    let res: number = (objSize.width / BASE_WIDTH) * appSize.width;
    return res;
};

const getWidthOf = (num: number) => {
    let res: number = (num / BASE_WIDTH) * 100;
    return res.toFixed(3) + '%';
};

const getWidthPxOf = (num: number) => {
    if (!appSize) {
        showError();
        return 0;
    }
    let res: number = (num / BASE_WIDTH) * appSize.width;
    return res;
};

const getHeight = () => {
    if (!objSize) {
        showError();
        return '0%';
    }
    let res: number = (objSize.height / BASE_HEIGHT) * 100;
    return res.toFixed(3) + '%';
};

const getHeightPx = () => {
    if (!objSize || !appSize) {
        showError();
        return 0;
    }
    let res: number = (objSize.height / BASE_HEIGHT) * appSize.height;
    return res;
};

const getHeightOfPx = (num: number) => {
    if (!appSize) {
        showError();
        return 0;
    }
    let res: number = (num / BASE_HEIGHT) * appSize.height;
    return res;
};

const getRelativeWidth = () => {
    if (!objSize || !appSize) {
        showError();
        return '0%';
    }
    let res: number = (objSize.width / BASE_HEIGHT / appSize.ratio) * 100;
    return res.toFixed(3) + '%';
};

const getRelativeHeight = () => {
    if (!objSize || !appSize) {
        showError();
        return '0%';
    }
    let res: number = (objSize.height / BASE_WIDTH) * appSize.ratio * 100;
    return res.toFixed(3) + '%';
};

const getStandard = (w: number, h: number, t: number) => {
    setObjSize(w, h);
    let res: {
        position: 'relative' | 'absolute' | undefined;
        width: string;
        height: string;
        left: number;
        top: number;
    } = {
        position: 'absolute',
        width: getWidth(),
        height: getHeight(),
        left: getCenterLeftPx(),
        top: getVerticalPx(t),
    };
    return res;
};

const getStandardPx = (w: number, h: number, t: number) => {
    setObjSize(w, h);
    let res: {
        position: 'relative' | 'absolute' | undefined;
        width: number;
        height: number;
        left: number;
        top: number;
    } = {
        position: 'absolute',
        width: getWidthPx(),
        height: getHeightPx(),
        left: getCenterLeftPx(),
        top: getVerticalPx(t),
    };
    return res;
};

const getPerfectPx = (w: number, h: number, l: number, t: number) => {
    setObjSize(w, h);
    let res: {
        position: 'relative' | 'absolute' | undefined;
        width: number;
        height: number;
        left: number;
        top: number;
    } = {
        position: 'absolute',
        width: getWidthPx(),
        height: getHeightPx(),
        left: getVerticalPx(l),
        top: getVerticalPx(t),
    };
    return res;
};

const getPosStaticHeight = (w: number, h: number, t: number) => {
    setObjSize(w, h);
    let widthObj = getWidthPx();
    let res: {
        position: 'relative' | 'absolute' | undefined;
        width: number;
        height: number;
        left: number;
        top: string;
    } = {
        position: 'absolute',
        width: getWidthPx(),
        height: (h / w) * widthObj,
        left: getCenterLeftPx(),
        top: getVertical(t),
    };
    return res;
};

const getOnlyPos = (w: number, h: number, t: number) => {
    setObjSize(w, h);
    let res: {
        position: 'relative' | 'absolute' | undefined;
        left: number;
        top: string;
    } = {
        position: 'absolute',
        left: getCenterLeftPx(),
        top: getVertical(t),
    };
    return res;
};

const getPosAndWid = (w: number, h: number, t: number) => {
    setObjSize(w, h);
    let res: {
        position: 'relative' | 'absolute' | undefined;
        width: string;
        left: number;
        top: string;
    } = {
        position: 'absolute',
        width: getWidth(),
        left: getCenterLeftPx(),
        top: getVertical(t),
    };
    return res;
};

const getPosWithMinHeight = (w: number, h: number, t: number, min: number) => {
    setObjSize(w, h);
    let heightObj = getHeightPx();
    let res: {
        position: 'relative' | 'absolute' | undefined;
        width: number;
        height: number;
        left: number;
        top: string;
    } = {
        position: 'absolute',
        width: getWidthPx(),
        height: height < min ? min : heightObj,
        left: getCenterLeftPx(),
        top: getVertical(t),
    };
    return res;
};

const fontRatio = width > 414 ? 1 : width / 414;
const getFontSize = (h: number) => h * fontRatio;
const mainButtonsHeight = (h: number) => h * fontRatio;

export {
    setAppSize,
    initAppSize,
    setObjSize,
    getCenterLeftPx,
    getCenterTopPx,
    getHorizontal,
    getHorizontalPx,
    getVertical,
    getVerticalPx,
    getWidth,
    getWidthOf,
    getWidthPx,
    getWidthPxOf,
    getHeight,
    getHeightPx,
    getHeightOfPx,
    getRelativeWidth,
    getRelativeHeight,
    getStandard,
    getStandardPx,
    getPerfectPx,
    getPosStaticHeight,
    getOnlyPos,
    getPosAndWid,
    getPosWithMinHeight,
    getStackHeaderHeight,
    getFontSize,
    mainButtonsHeight,
};
