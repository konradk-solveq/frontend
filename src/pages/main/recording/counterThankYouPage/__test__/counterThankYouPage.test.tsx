import 'react-native';
import React from 'react';
import renderComponent from '@jestUtils/render';
import asyncEvent from '@jestUtils/asyncEvent';
import {initAppSize} from '@helpers/layoutFoo';
import CounterThankYouPage from '@pages/main/recording/counterThankYouPage/counterThankYouPage';

const TEST_TIME = 1800000; // 30 min
const TEST_PAUSE = 300000; // 5 min
const TEST_DISTANCE = '10,00';
const TEST_NAME = 'Test';
const TEST_PARSED_TIME = '0:25';
const TEST_PARSED_PAUSE = '0:05';

jest.mock('react', () => {
    const originReact = jest.requireActual('react');
    const mUseRef = jest.fn();
    return {
        ...originReact,
        useRef: mUseRef,
    };
});

jest.mock('@react-navigation/native', () => ({
    useNavigation: () => ({
        canGoBack: jest.fn().mockReturnValue(true),
    }),
    useRoute: () => ({
        params: {
            time: TEST_TIME,
            pause: TEST_PAUSE,
            distance: TEST_DISTANCE,
        },
    }),
}));

const mockedGetName = jest.fn();

const initStore = {
    routes: {
        loading: false,
        currentRoute: {
            pauseTime: TEST_PAUSE,
        },
        error: '',
        statusCode: 200,
    },
    user: {
        userName: TEST_NAME,
    },
};

describe('Thank You Counter Screen', () => {
    const mockedNavigation = {
        navigate: jest.fn(),
    };

    beforeAll(() => {
        initAppSize();
    });

    describe('Rendering', () => {
        it('should match snapshot', async () => {
            const component = await asyncEvent(
                renderComponent(
                    <CounterThankYouPage
                        navigation={mockedNavigation}
                        getName={mockedGetName}
                        name={TEST_NAME}
                    />,
                    undefined,
                    initStore,
                ),
            );

            expect(component).toMatchSnapshot();
        });
        afterEach(() => {
            jest.restoreAllMocks();
        });
    });
});
