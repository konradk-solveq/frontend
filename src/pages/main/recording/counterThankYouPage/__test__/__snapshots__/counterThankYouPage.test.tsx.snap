// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thank You Counter Screen Rendering should match snapshot 1`] = `null`;
